### PR TITLE
PMREMGenerator: Remove calls of convertSRGBToLinear().

### PR DESCRIPTION
**Pull request by Mugen87**
Related issue: see https://github.com/mrdoob/three.js/issues/22286#issuecomment-896792847

**Description**

~~`PMREMGenerator` now honors `WebGLRenderer.outputEncoding` when converting to linear color space.~~

This PR removes calls of `convertSRGBToLinear()`. Background/clear colors are assumed to be in linear color space.

### CODE REVIEW

**Comment by Mugen87** [comment]
`webgl_furnace_test` assumes sRGB for `Scene.background`.

https://github.com/mrdoob/three.js/blob/f4601f33b222c34dedb4b2d4d5dd554894fab251/examples/webgl_furnace_test.html#L113

That means it is now necessary to set `WebGLRenderer.outputEncoding` to `sRGBEncoding` which slightly changes its visuals.

**Review by WestLangley — CHANGES_REQUESTED** [review]
>That means it is now necessary to set WebGLRenderer.outputEncoding to sRGBEncoding

No, that does not follow. The furnace test can be rendered in any colorspace. I'd remove the changes to the furnace test.

I think it would be best if @gkjohnson reviews the PMREM edits in this PR.

**Comment by Mugen87** [comment]
I've edited the example since `Scene.background` is explicitly defined as a sRGB color (see comment).

**Comment by WestLangley** [comment]
>I've edited the example since Scene.background is explicitly defined as a sRGB color (see comment).

I wrote the comment, and the comment is referring to the fact that `PMREMGenerator` is assuming the color is sRGB. This PR supposedly fixes that limitation of `PMREMGenerator`.

The furnace test is broken until [other issues](https://github.com/mrdoob/three.js/issues/22286#issuecomment-896503667) with `PMREMGenerator` are fixed.

**Comment by gkjohnson** [comment]
The changes to PMREMGenerator look good -- I'm not sure if any other types of output encodings are ever used for the renderer but my only comment is if only SRGB and Linear are supported and other output encodings might be used by users (not sure why they would be) you might want to log a warning.

I'm not familiar enough with the furnace test or the interpretation of RGBE colors in the shaders to be much help there, though. `@elalish` might be a better one to ping when you start looking into that more.

**Comment by Mugen87** [comment]
>  I'm not sure if any other types of output encodings are ever used for the renderer

Um, I guess I would add support for `THREE.GammaEncoding`. The conversion can be done via `convertGammaToLinear()`. For all other constants, a warning should be logged.

**Comment by Mugen87** [comment]
With the last commit I've added support for `THREE.GammaEncoding`. `PMREMGenerator` also logs a warning now when color spaces other than linear, gamma and srgb are used.

**Comment by WestLangley** [comment]
@Mugen87 @gkjohnson Sorry, but this PR is not correct.

`PMREMGenerator.fromScene()` was written to create a PMREM from something like `RoomEnviornment`. The current renderer's output colorspace is _irrelevant_ in that case. `envScene` background color should be _specified_ in linear space because the renderer will render the PMREM in linear space. The background color should not be converted by PMREM.

If @gkjohnson is creating a PMREM from his current scene, that would be different use case.

**Comment by Mugen87** [comment]
> The background color should not be converted by PMREM.

But in this case, any calls of `convertSRGBToLinear()` should be removed from `PMREMGenerator`, right?

**Comment by WestLangley** [comment]
>But in this case, any calls of convertSRGBToLinear() should be removed from PMREMGenerator, right?

Yes.

And as I said, sRGB output encoding in `webgl_furnace_test.html` is optional, and should be noted as such. The conversion to linear (if needed) belongs at the app level.

**Comment by Mugen87** [comment]
Changed the title of the PR and updated the code.

File: examples/webgl_furnace_test.html [hunk 1]
```diff
@@ -110,14 +110,13 @@
 				return new Promise( function ( resolve ) {
 
 					const envScene = new THREE.Scene();
-					envScene.background = new THREE.Color( COLOR ); // sRGB colorspace assumed, apparently
+					envScene.background = new THREE.Color( COLOR );
```
Comment (WestLangley) [inline examples/webgl_furnace_test.html:113]: You need to add this:
```js
if ( renderer.outputEncoding === THREE.sRGBEncoding ) envScene.background.convertSRGBToLinear();
```

Comment (Mugen87) [reply examples/webgl_furnace_test.html:113]: Done!

**Comment by gkjohnson** [comment]
> If @gkjohnson is creating a PMREM from his current scene, that would be different use case.

I see what you're saying. The couple times I've used PMREMGenerator it's been to render my current scene in order to get reflections on objects in that scene. It seems it would be difficult to simply support both use cases here.

This comes back to something like #22275 and / or bookkeeping color space information on the Color class so the renderer has the relevant information to interpret the colors correctly.

**Review by WestLangley — APPROVED** [review]

**Comment by WestLangley** [comment]
Merging for now so #22326 can be fixed.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**